File: app/src/main/java/com/igordanilchik/daggertest/api/ClientApi.java
package com.igordanilchik.daggertest.api;


import com.igordanilchik.daggertest.model.Catalogue;

import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

public interface ClientApi {

    String API_BASE_URL = "http://ufa.farfor.ru";
    String API_KEY = "ukAXxeJYZN";

    @GET("/getyml")
    Observable<Catalogue> loadCatalogue(@Query("key") String key);
}

File: app/src/main/java/com/igordanilchik/daggertest/api/ClientApi.kt
package com.igordanilchik.daggertest.api


import com.igordanilchik.daggertest.dto.inner.Catalogue

import retrofit2.http.GET
import retrofit2.http.Query
import rx.Observable

interface ClientApi {

    @GET("/getyml")
    fun loadCatalogue(@Query("key") key: String): Observable<Catalogue>

    companion object {

        val API_BASE_URL = "http://ufa.farfor.ru"
        val API_KEY = "ukAXxeJYZN"
    }
}
